lib/Test/Stream/Formatter/TAP.pm

```diff
@@ -99,6 +116,193 @@ sub _autoflush {
     select $old_fh;
 }
 
+sub event_tap {
+    my $self = shift;
+    my ($e, $num) = @_;
+
+    my $converter = $CONVERTERS{ref($e)};
+
+    $num = undef if $self->{+NO_NUMBERS};
+
+    # Legacy Support for $e->to_tap
+    unless ($converter) {
+        my $legacy = $e->can('to_tap') or return;
+        return if $legacy == \&Test::Stream::Event::to_tap;
+        warn "'$e' implements 'to_tap'. to_tap methods on events are deprecated.\n";
+        return $e->to_tap($num);
+    }
+
+    return $self->$converter($e, $num);
+}
+
+sub _ok_event {
+    my $self = shift;
+    my ($e, $num) = @_;
+
+    # We use direct hash access for performance. OK events are so common we
+    # need this to be fast.
+    my $name  = $e->{name};
+    my $debug = $e->{debug};
+    my $todo  = $e->{todo};
```

Comment:
I did some microbenchmarks:

Here is the code that was run:

https://gist.github.com/kentfredric/3d8d862656ba33a60a74

And here's the data that was gleaned:

https://docs.google.com/spreadsheets/d/1IFVXHAkVlM2rYMl8HqDzLiOgoJRxb_KXUFFweaiRyns/pubhtml?gid=498365932&single=true

![time sequence](https://docs.google.com/spreadsheets/d/1IFVXHAkVlM2rYMl8HqDzLiOgoJRxb_KXUFFweaiRyns/pubchart?oid=1032411648&format=image)

![histogram](https://docs.google.com/spreadsheets/d/1IFVXHAkVlM2rYMl8HqDzLiOgoJRxb_KXUFFweaiRyns/pubchart?oid=1214578820&format=image)

I think based on that data, `op1` is the faster of the two, and `op1` happens to be the parallel-assign case.

But as to its effects at scale and as to if there's any errors in my methodology or analysis, that's a different question.

And as to why there's a creep-up effect over time, that's also something worth investigating O.o